test/unit-test/ota_utest.c

```diff
@@ -658,10 +612,9 @@ static void otaReceiveJobDocument()
     OTA_SignalEvent( &otaEvent );
 }
 
-/* Jump to any state in OTA agent. Event send interface must be set to mockOSEventSendThenStop to
- * prevent OTA internal state transition. */
-static void otaGoToStateWithTimeout( OtaState_t state,
-                                     int timeout_ms )
+/* Jump to any state in OTA agent. The event send interface must be set to
+ * mockOSEventSendThenStop to prevent any OTA internal state transitions. */
+static void otaGoToState( OtaState_t state )
```

Comment:
Is there a reason wait with timeout was removed?  Shouldn't we wait for the event handler to finish in a specific time?

Reply:
It was removed because the code is no longer multithreaded so it's not needed. The code will go to the given state and then return. So there is no need to wait.